libs/application/templates/european-health-insurance-card/src/lib/helpers/applicantHelper.ts

```diff
@@ -0,0 +1,238 @@
+/** Returns name of person by national registry id by external data from National Registry */
+export function getFullName(
+  formValues: Application<FormValue>,
+  nrid: string | null | undefined,
+) {
+  if (nrid) {
+    const persons = getFromRegistry(formValues)
+    return persons.find((x) => x.nrid === nrid)?.name
+  }
+  return null
+}
+
+export function getEhicApplicants(
+  formValues: Application<FormValue>,
+  cardType: string | null,
+) {
+  const nridArr: string[] = []
+  const userData = formValues.externalData.nationalRegistry
+    ?.data as NationalRegistry
+  if (userData?.nationalId) {
+    nridArr.push(userData?.nationalId)
+  }
+  const spouseData = formValues?.externalData?.nationalRegistrySpouse
+    ?.data as NationalRegistry
+
+  if (spouseData?.nationalId) {
+    nridArr.push(spouseData?.nationalId)
+  }
+
+  const custodyData = (formValues?.externalData
+    ?.childrenCustodyInformation as unknown) as NationalRegistry
+
+  if (custodyData?.nationalId) {
+    nridArr.push(custodyData?.nationalId)
+  }
+
+  if (!cardType) {
+    return nridArr
+  }
+
+  const applicants = getObjectKey(formValues.answers, true)
+  const applying: string[] = []
+
+  if (applicants.includes(`${cardType}-${userData?.nationalId}`)) {
+    applying.push(userData?.nationalId)
+  }
+
+  if (applicants.includes(`${cardType}-${spouseData?.nationalId}`)) {
+    applying.push(spouseData?.nationalId)
+  }
+
+  for (let i = 0; i < custodyData.data.length; i++) {
+    if (applicants.includes(`${cardType}-${custodyData.data[i].nationalId}`)) {
+      applying.push(custodyData.data[i])
+    }
+  }
+
+  return applying
+}
+
+export function base64ToArrayBuffer(base64Pdf: string) {
+  const binaryString = window.atob(base64Pdf)
+  const binaryLen = binaryString.length
+  const bytes = new Uint8Array(binaryLen)
+  for (let i = 0; i < binaryLen; i++) {
+    const ascii = binaryString.charCodeAt(i)
+    bytes[i] = ascii
+  }
+  return bytes
+}
+
+export function someCanApplyForPlasticOrPdf(
+  externalData: ExternalData,
+): boolean {
+  if (externalData?.cardResponse?.data) {
+    const cardResponse = externalData?.cardResponse?.data as CardResponse[]
+    for (let i = 0; i < cardResponse?.length; i++) {
+      if (cardResponse[i].isInsured && cardResponse[i].canApply) {
+        return true
+      }
+
+      if (cardResponse[i].isInsured && !cardResponse[i].canApply) {
+        const tempCard = cardResponse[i].cards?.find((x) => x.isTemp)
+        if (tempCard) {
+          continue
+        }
+        const plasticCard = cardResponse[i].cards?.find((x) => x.isPlastic)
+        if (plasticCard) {
+          return true
+        }
+      }
+    }
+  }
+  return false
+}
+
+/** Checks if some are insured*/
+export function someAreInsured(externalData: ExternalData): boolean {
+  if (externalData?.cardResponse?.data) {
+    const cardResponse = externalData?.cardResponse?.data as CardResponse[]
+
+    const ret = cardResponse.find((x) => x.isInsured === true)
```

Comment:
If you use the some() function instead of find() you can directly return a boolean value. The some() function returns true if at least one element in the array satisfies the given condition, and false otherwise.